### ISSUE TITLE
`exclude()` with relationship in query breaks search in django 3.2+

### ISSUE THREAD

**Issue by danthedeckie** [wagtail/wagtail#7828]
### Issue Summary

Querysets with `.exclude(<relationset_name>=None)` cannot be used with `.search()` as it throws a `AttributeError: 'Exists' object has no attribute 'target'` error.

For example:

```python
>>> Page.objects.exclude(revisions=None).search('Home')
...wagtail/search/backends/base.py in _get_filters_from_where_node(self, where_node, check_only)
    100                     )
    101             else:
--> 102                 field_attname = where_node.lhs.target.attname
    103             lookup = where_node.lookup_name
    104             value = where_node.rhs

AttributeError: 'Exists' object has no attribute 'target'
```

Using `filter(<relationset_name>__isnull=True)` seems to work correctly.

### Steps to Reproduce

1. Start a new project with `wagtail start myproject`
2. `cd myproject`
2. `./manage.py shell`
3. `from wagtail.core.models import Page;Page.objects.exclude(revisions=None).search('hello')`

* I have confirmed that this issue can be reproduced as described on a fresh Wagtail project: (yes)

### Technical details

* Python version: 3.8, 3.9, 3.10
* Django version: 3.2 (3.1 seems to be fine!)
* Wagtail version: 2.14, 2.15
* Browser version: N/A

### Possibly relevant links:
Was this related to the cause?
https://github.com/django/django/pull/13606
A possibly similar fix:
https://github.com/wagtail/wagtail/issues/5967

**Comment by satyam-jha** [comment]
Hi, is this issue still open? I like to contribute

**Comment by gasman** [comment]
@satyam-jha Yes, go ahead!

**Comment by TUSHAR-VERMA-star** [comment]
@satyam-jha are you working on this issue ?

**Comment by satyam-jha** [comment]
@TUSHAR-VERMA-star you can pick up this issue if you want, as I wasn't able to actually find the issue

**Comment by TUSHAR-VERMA-star** [comment]
@satyam-jha  Ok

**Comment by TUSHAR-VERMA-star** [comment]
@gasman can I take this issue ?

**Comment by gasman** [comment]
@TUSHAR-VERMA-star Yep, go ahead! (Generally there's no need to ask permission - if there's an open issue with no existing fix and no sign of people working on it, you're OK to pick it up.)

**Comment by TUSHAR-VERMA-star** [comment]
Ok Done

**Comment by gasman** [comment]
Closing as a duplicate of wagtail/wagtailsearch#9, as per https://github.com/wagtail/wagtail/pull/10012#issuecomment-3067433065.
